tests/hap/test_svm_j97e06.py

```diff
@@ -0,0 +1,238 @@
+
+    print("\ngather_data_for_processing. Gathered data: {}".format(files_to_process))
+
+    return files_to_process
+
+
+@pytest.fixture(scope="module")
+def gather_output_data(construct_manifest_filename):
+    # Determine the filenames of all the output files from the manifest
+    files = []
+    with open(construct_manifest_filename, 'r') as fout:
+        for line in fout.readlines():
+            files.append(line.rstrip("\n"))
+    print("\ngather_output_data. Output data files: {}".format(files))
+
+    return files
+
+
+@pytest.fixture(scope="module")
+def construct_manifest_filename(read_csv_for_filenames):
+    # Construct the output manifest filename from input file keywords
+    inst = fits.getval(read_csv_for_filenames[0], "INSTRUME", ext=0).lower()
+    root = fits.getval(read_csv_for_filenames[0], "ROOTNAME", ext=0).lower()
+    tokens_tuple = (inst, root[1:4], root[4:6], "manifest.txt")
+    manifest_filename = "_".join(tokens_tuple)
+    print("\nconstruct_manifest_filename. Manifest filename: {}".format(manifest_filename))
+
+    return manifest_filename
+
+
+@pytest.fixture(scope="module", autouse=True)
+def svm_setup(gather_data_for_processing):
+    # Act: Process the input data by executing runsinglehap - time consuming activity
+
+    current_dt = datetime.datetime.now()
+    print(str(current_dt))
+    print("\nsvm_setup fixture")
+
+    # Read the "poller file" and download the input files, as necessary
+    input_names = gather_data_for_processing
+
+    # Run the SVM processing
+    path = os.path.join(os.path.dirname(__file__), POLLER_FILE)
+    try:
+        status = runsinglehap.perform(path)
+
+    # Catch anything that happens and report it.  This is meant to catch unexpected errors and
+    # generate sufficient output exception information so algorithmic problems can be addressed.
+    except Exception as except_details:
+        print(except_details)
+        pytest.fail("\nsvm_setup. Exception Visit: {}\n", path)
+
+    current_dt = datetime.datetime.now()
+    print(str(current_dt))
+
+
+# TESTS
+
+def test_svm_manifest_name(construct_manifest_filename):
+    # Construct the manifest filename from the header of an input file in the list and check it exists.
+    path = Path(construct_manifest_filename)
+    print("\ntest_svm_manifest. Filename: {}".format(path))
+
+    # Ensure the manifest file uses the proper naming convention
+    assert(path.is_file())
+
+
+def test_svm_wcs(gather_output_data):
+    # Check the output primary WCSNAME includes FIT_SVM_GAIA as part of the string value
+    tdp_files = [files for files in gather_output_data if files.lower().find("total") > -1 and files.lower().endswith(".fits")]
+
+    for tdp in tdp_files:
+        wcsname = fits.getval(tdp, "WCSNAME", ext=1).upper()
+        print("\ntest_svm_wcs.  WCSNAME: {} Output file: {}".format(wcsname, tdp))
+        assert WCS_SUB_NAME in wcsname, f"WCSNAME is not as expected for file {tdp}."
```

Comment:
Will this test stop immediately (not evaluate any more files) on the FIRST instance where the assert fails?  Is this the best behavior for this type of test?  This comment also applies to the rest of the tests here as well.

Reply:
Fixed so the assert is not done in a loop in any of the tests.